### ISSUE TITLE
About visivility of NIP-36 (content-warning) specified content

### ISSUE THREAD

**Issue by ryogrid** [nostr-protocol/nips#769]
# Problem
- Users specify content-waring to own post for making not visible it to people who don't approbed that it is sensitive content
- But current NIP-36 specification can't realize above because not NIP-36 compliant clients such as Damus shows all kind-1 event's content without noticing the content-warning marking
- Lack of consideration described above left users with a psychological barrier that makes it difficult to post sensitive content

# Solution
- Modify NIP-36 specification to use kind number for content-warning specified event instead of 1
- Or, define NIP-36 extended new NIP like above

**Comment by AsaiToshiya** [comment]
I agree with need for this proposal. Probably we need to think about some kinds not only kind 1.

**Comment by AsaiToshiya** [comment]
What about the solution to wrap the event?

For example:

```json
{
    "pubkey": "<pub-key>",
    "created_at": 1000000000,
    "kind": 30036,
    "tags": [
      ["d", "<id of the wrapped event>"],
      ["k", "1"],
      ["L", "content-warning"],
      ["l", "reason", "content-warning"],
      ["L", "social.nos.ontology"],
      ["l", "NS-nud", "social.nos.ontology"],
      ["content-warning", "reason"] /* reason is optional */
    ],
    "content": "<stringified event, e.g. kind 1>",
    "id": "<event-id>"
}
```

**Comment by ryogrid** [comment]
@AsaiToshiya 
Thanks for comment and suggestion :)

> What about the solution to wrap the event?

Your idea looks nice.
I understund this as technique like NIP of repost (NIP-18) .

I have a question.
- Your suggested idea assumes that wrapped event is not sent to relay in not wrapped form?
  - (If answer of above is Yes, I think that suggested idea can solve the iessue)

**Comment by AsaiToshiya** [comment]
> Your idea looks nice. I understund this as technique like NIP of repost (NIP-18) .

Ah yes right.

> I have a question.
> 
> * Your suggested idea assumes that wrapped event is not sent to relay in not wrapped form?
>   
>   * (If answer of above is Yes, I think that suggested idea can solve the iessue)

Yes, the wrapped event itself is not to be sent to relays.

But, my solution have a problem that cannot filter by d tag value of the wrapped event.

**Comment by SnowCait** [comment]
How about copying tags which are expected to be filtered (= `#<single-letter>`) to kind 30036?

**Comment by AsaiToshiya** [comment]
Yes, it is the only way.

**Comment by AsaiToshiya** [comment]
@fernandolguevara Please give us your opinion as NIP-36 author.

**Comment by AsaiToshiya** [comment]
Support for `kind 30036` have been added to Brostr: https://github.com/AsaiToshiya/brostr/commit/6b41e47e00372892517c38ab0c00a39ddef10a96

Example: https://asaitoshiya.github.io/brostr/src?nip19=nevent1qqs0kucgkfwnevn7gtdlt2huea74emccc2qza53fc90n978lmhx73yc0e4yzs

**Comment by staab** [comment]
I don't like this approach. It introduces unnecessary complexity. I don't see why we need to force clients to hide content with a warning by default. That should be up to the client developer. I don't think this is going to gain wide support.

**Comment by AsaiToshiya** [comment]
I have the opposite opinion. Clients that don't support it, don't have to show content that users want to hide by default. Whether or not to support it is also up to the client developer.

**Comment by gpsnmeajp** [comment]
I am deeply interested in this issue.

I understand that NIP-36 exists for the purpose of allowing senders to issue warnings to recipients. For instance, it's used to voluntarily indicate content that requires viewer discretion, such as information inappropriate for children, warnings for epilepsy patients, or horror content.

In many cases, the sender likely does not want to show content to recipients who do not agree with the warnings.

It's a user's prerogative to choose any client, and certainly, it's up to the clients whether to implement NIP-36 or not. There should be no issue if users intentionally choose clients that intentionally do not support warnings.

However, since NIP-36 has been defined as an additional attribute for kind1, the current situation is such that most clients used by the majority simply haven't implemented it, resulting in content being displayed without any warnings.

Considering the intent behind using NIP-36, there's no point in employing this attribute if the majority of people are viewing the content without warnings. Similarly, there's no rationale for clients to implement a feature that's not being used currently.

In such a scenario, the existence of NIP-36 becomes redundant.

Therefore, I agree with the proposal in this issue to "make content not appear at all in clients that do not support it." This would give a meaningful reason to choose clients that intentionally ignore warnings, and also make it possible for users to avoid such clients.

**Comment by staab** [comment]
> the current situation is such that most clients used by the majority simply haven't implemented it

This is false, Coracle, Amethyst, Snort, and Nostrudel all support it. And like you said, people can choose what clients they use.

**Comment by mikedilger** [comment]
I support this idea. But I don't think you should stringify a kind-1 into the contents, but just make this another kind of feed-related event that current clients won't know about until they program for it.

But like @staab says, this may not gain wide support. And there is also no way to know if a devious client coded for this by always just showing the event fully opened up, along with the warning, for a lark... that is to say, you can't force the hand of client developers.  For example, gossip doesn't actually delete events, it renders them with strikethrough.  So this can only be truly solved client-side, it cannot be fixed via the protocol.

Even so, I support having another event kind that does this if people want it and if there are 2 clients using it.

BTW: gossip supports content-warnings as they are currently.
